src/refda/adm/ietf_dtnma_agent.c

```diff
@@ -299,6 +339,258 @@ static cace_ari_vast bitwise_xor_vast(cace_ari_vast left, cace_ari_vast right)
     return left ^ right;
 }
 
+static int timespec_numeric_add(cace_ari_t *result, const cace_ari_t *valueA, const cace_ari_t *valueB)
+{
+    CHKERR1(result);
+    CHKERR1(valueA);
+    CHKERR1(valueB);
+
+    cace_ari_type_t typeA = eqiv_ari_type(&(valueA->as_lit));
+    cace_ari_type_t typeB = eqiv_ari_type(&(valueB->as_lit));
+
+    // Addition of TD and TD results in a TD value
+    bool is_result_TD = false;
+    is_result_TD |= typeA == CACE_ARI_TYPE_TD && typeB == CACE_ARI_TYPE_TD;
+
+    // Addition of TP and TD results in a TP value
+    bool is_result_TP = false;
+    is_result_TP |= typeA == CACE_ARI_TYPE_TD && typeB == CACE_ARI_TYPE_TP;
+    is_result_TP |= typeA == CACE_ARI_TYPE_TP && typeB == CACE_ARI_TYPE_TD;
+
+    // Determine the cace_ari_type_t of the result
+    cace_ari_type_t ari_type;
+    if (is_result_TD == true)
+        ari_type = CACE_ARI_TYPE_TD;
+    else if (is_result_TP == true)
+        ari_type = CACE_ARI_TYPE_TP;
+    else
+        // Bail if the result will not be a TD or TP
+        return 1;
+
+    // Delegate the addition
+    struct timespec valueA_TS = valueA->as_lit.value.as_timespec;
+    struct timespec valueB_TS = valueB->as_lit.value.as_timespec;
+    struct timespec result_TS = timespec_add(valueA_TS, valueB_TS);
+
+    // Store the result
+    cace_ari_deinit(result);
+    cace_ari_lit_t *res_lit = cace_ari_init_lit(result);
+
+    res_lit->has_ari_type      = true;
+    res_lit->ari_type          = ari_type;
+    res_lit->prim_type         = CACE_ARI_PRIM_TIMESPEC;
+    res_lit->value.as_timespec = result_TS;
+
+    // Success
+    return 0;
+}
+
+static int timespec_numeric_sub(cace_ari_t *result, const cace_ari_t *valueA, const cace_ari_t *valueB)
+{
+    CHKERR1(result);
+    CHKERR1(valueA);
+    CHKERR1(valueB);
+
+    cace_ari_type_t typeA = eqiv_ari_type(&(valueA->as_lit));
+    cace_ari_type_t typeB = eqiv_ari_type(&(valueB->as_lit));
+
+    // Note the following will result in TD:
+    // - Subtraction of TP from TP
+    // - Subtraction of TD from TD
+    bool is_result_TD = false;
+    is_result_TD |= typeA == CACE_ARI_TYPE_TP && typeB == CACE_ARI_TYPE_TP;
+    is_result_TD |= typeA == CACE_ARI_TYPE_TD && typeB == CACE_ARI_TYPE_TD;
+    if (is_result_TD == false)
+        // Bail if the result will not be a TD
+        return 1;
+
+    struct timespec valueA_TS = valueA->as_lit.value.as_timespec;
+    struct timespec valueB_TS = valueB->as_lit.value.as_timespec;
+
+    //    fprintf(stdout, "\nDEBUG--->   [A]: (%ld sec, %ld nano)   ", valueA_TS.tv_sec, valueA_TS.tv_nsec);
```

Comment:
It is possible to keep these as `CACE_LOG_DEBUG` statements if they will be useful, otherwise I recommend deleting the dead code.